### DISCUSSION TITLE
xmltv guide

### DISCUSSION THREAD

**Discussion by kogren161** [tgorgdotcom/locast2plex#219] [Q&A]
I am back to trying to run Locast2Plex in Linux.  I have setup a raspberry pi 4 with Jellyfin, which all works, except the guide.  I am still having trouble getting the guide data file.  I have been trying to run the mc2xml file from the command line, but I get command not found.  I jumped through a bunch of hoops trying to get the zap2xml to run, but it seems to have problems now in windows and Linux.  I can generate the xmltv file in windows with mc2xml.pl.  Does anyone have any suggestions?

**Comment by libersongm** [comment]
I got mc2xml to work.  Will send all I have in AM.  I thought I am using executable, cannot remember.  Let you know in AM.  understand PI is ARM.  

Gary

**Reply by kogren161** [reply]
Thanks, I'd really appreciate it.

**Reply by libersongm** [reply]
Sorry been delayed, weather has been an issue in DFW.   Power, Super Cold for the area, etc.  Will send as soon as able.   I never got the peel stuff to work, only the executable worked.  Haven't seen an ARM version only x86.   When I can will let you know

**Reply by libersongm** [reply]
I am using the compiled version of mc2xml.   Here is the file attribute:
mc2xml: ELF 32-bit LSB executable, Intel 80386, version 1 (GNU/Linux), statically linked, no section header

I was using TV guide but recently rain into problems.   I run a cron job at 2:00am

Here is my call:  /home/directory/xmltv/mc2xml -1 -c us -g zipcode -d 336 -D mc2xml.dat -o xmltv.xml

Directory is your home login name

I haven't found an ARM version of the compiled code.   Do you have the code and could compile?

Thanks,
Gary

**Reply by libersongm** [reply]
Ok was testing out zap2epg, a kodi addon in GITHUB.  https://github.com/edit4ever/script.module.zap2epg

It seems to be a python2 script that runs and uses zap2it for data.   It is setup as Kodi addon.  But tinkering with the python code, and yes I use tinker, I am an old perl hack, not a python guy, but I can load in a zipcode and a days look ahead and I can get it to command run the python script.   You have to set permissions wide open on the files for Jellyfin.   I am testing now, if I make progress, I will upload code hacks and how to script if you would like to try out on ARM.  I am using mc2xml for i386.

Ok here is how it works.   Extract the download zip to a directory.   chmod on the directory and files to wide open.  Run the basic script using python2.   See if it works.  It downloads I believe 24 hours of guide for zipcode 92101.

run python2 on the zap2epg.py or the hacked version.   I am not sure the cache is working right.  I have to frequently blow away the cache.   Here is a diff for the hacked version where I put in a zipcode where xxxxx is your US zipcode numeric.   Days can be what every look ahead number up to 14 days.  I am sorry just was testing.  See if this does anything for you.

Here is the diff

57c57,58
<             zipcode = settingsDict[setting]
---
> #            zipcode = settingsDict[setting]
>             zipcode = 'xxxxx'
65c66,67
<             days = settingsDict[setting]
---
> #            days = settingsDict[setting]
>             days = '7'

It seems to work.  If I was a better python hack I would make it much better, was just testing to see if it worked.

Very Respectfully,
Gary

**Reply by kogren161** [reply]
Thanks so much.

I hope you are doing ok with the Wild weather in Texas.
I appreciate your help, and will be interested in the results of your 'tinkering'

**Reply by libersongm** [reply]
Thanks for asking.  All good was 71F today.  So recommended downloading project zap2epg, then change the two lines, zip code and hours and try it out.  Make sense?  Let me know how it goes.
